EnhancedMediaPlayer/MediaPlayerView.swift

```diff
@@ -82,27 +85,79 @@ public struct MediaPlayerView: View {
             .wrapInButton(action: { viewModel.onTapAction?(control) })
     }
     
-    @ViewBuilder private func renderTimeElapsedWrapper<Content: View>(
+    @ViewBuilder private func renderTimePlaceholderWrapper<Content: View>(
         @ViewBuilder content: @escaping () -> Content
     ) -> some View {
         ZStack {
             content()
-            Text("10") // TODO: update with amount of rewind / forward
+            Text(String(format: "%.0f", viewModel.seekFactor))
                 .foregroundColor(.white)
                 .font(.system(size: Constants.timeFontSize))
         }
     }
+
+    @ViewBuilder private func renderSeekTappableScreen() -> some View {
+        HStack(spacing: .zero) {
+            renderTappableScreen(control: .rewind, seek: .rewindTap)
+            renderTappableScreen(control: .forward, seek: .forwardTap)
+        }
+    }
+
+    @ViewBuilder private func renderTappableScreen(control: Control, seek: Seek) -> some View {
+        ZStack {
+            Rectangle()
+                .foregroundColor(.clear)
+                .contentShape(Rectangle())
+                .onTapGesture(count: Constants.seekTaps) {
+                    viewModel.onTapAction?(control)
+                    viewModel.onSeekDoubleTap?()
+                    renderSeekTaps(seek)
+                }
+            Rectangle()
+                .foregroundColor(.clear)
+                .contentShape(Rectangle())
+                .allowsHitTesting(viewModel.isTappingSeek)
+                .onTapGesture {
+                    viewModel.onTapAction?(control)
+                    viewModel.onSeekDoubleTap?()
+                    renderSeekTaps(seek)
+                }
```

Comment:
These two `Rectangles` look weird to me. The only difference between them is the `.allowsHitTesting()` modifier. How are they different, and why couldn't we move the `.allowHisTesting()` to the first one and remove the second?

Reply:
ditto, this is pretty much duplicated code - could you digress over the reason why? and if not, could you extract it to a function?

Reply:
I feel like this is hard to understand, so I'll refactor the code to make it cleaner. But here's the situation this rectangles handle.

- A user double taps a screen: the first Rectangle tracks this double tap and forward the video
- Right after the double tap, a user give one more tap: the second Rectangle tracks this single tap and forward the video
- `allowsHitTesting` is used to enable forward (or rewind) single taps only right after a double tap. If not used, the user can`t single tap the screen to show normal button controls